### PR TITLE
Enabling Decommission

### PR DESCRIPTION
**Pull request by chrislovecnm**
- enabled deco via changing feature gate
- changed unit test, to test that it is enabled
- updated operator_features comments
- moved prune pvc to alpha
- we have two e2e tests, one that prunes PVCs and another that
  does not prune pvcs.

### CODE REVIEW

File: e2e/decomission/decomission_test.go [hunk 1]
```diff
@@ -49,15 +49,68 @@ func TestMain(m *testing.M) {
...
 // It then checks that the cluster is stable and that decomissioning is successful.
 func TestDecommissionFunctionality(t *testing.T) {
 
 	// Testing removing and decommisioning a node.  We start at 4 node and then
 	// remove the 4th node
 
-	// turn on featuregate since Decommission is disabled by default currently
-	utilfeature.DefaultMutableFeatureGate.Set("UseDecommission=true,AutoPrunePVC=true")
+	// making sure the feature gate is off for prunePVC
+	utilfeature.DefaultMutableFeatureGate.Set("AutoPrunePVC=false")
```
Comment (udnay) [inline e2e/decomission/decomission_test.go:113]: On this test can we check to make sure there is an orphaned PV?

File: e2e/decomission/decomission_test.go [hunk 2]
```diff
@@ -49,15 +49,68 @@ func TestMain(m *testing.M) {
...
+			Test: func(t *testing.T) {
+				current := builder.Cr()
+				require.NoError(t, sb.Get(current))
+
+				current.Spec.Nodes = 3
+				require.NoError(t, sb.Update(current))
+				testutil.RequireClusterToBeReadyEventuallyTimeout(t, sb, builder, 500*time.Second)
+				testutil.RequireDecommissionNode(t, sb, builder, 3)
+				testutil.RequireDatabaseToFunction(t, sb, builder)
+				t.Log("Done with decommision")
```
Comment (udnay) [inline e2e/decomission/decomission_test.go:98]: The opposite here, that no PVC remains

Comment (chrislovecnm) [reply e2e/decomission/decomission_test.go:98]: What am I missing?

Comment (udnay) [reply e2e/decomission/decomission_test.go:98]: I think the logic doesn't immediately prune the PVC so we probably can't check for this anyhow. What I was thinking of was something like `assert.Equal(t, numPVCsForCRDB, 3)` or something. The `RequireDecommissionNode` checks CRDB to validate the operation is completed but doesn't check the orphaned PVC

Comment (udnay) [reply e2e/decomission/decomission_test.go:98]: I don't think it's testable here tbh.

**Review by udnay — APPROVED** [review]
LGTM

**Pull request merged**